yt_dlp/extractor/theplatform.py

```diff
@@ -295,6 +302,17 @@ def _real_extract(self, url):
 
         formats, subtitles = self._extract_theplatform_smil(smil_url, video_id)
 
+        # With some sites, manifest URL must be forced to extract HLS formats
+        if not traverse_obj(formats, lambda _, v: v['format_id'].startswith('hls')):
+            m3u8_url = update_url(url, query='mbr=true&manifest=m3u', fragment=None)
+            ext = urlhandle_detect_ext(self._request_webpage(
+                HEADRequest(m3u8_url), video_id, 'Checking for additional m3u8 formats', fatal=False))
```

Comment:
so `urlhandle_detect_ext(self._request_webpage(...))` is fatal, needs to be fixed

also adjust the errnote so that the test's expected warning can be '`No HLS formats found`'
```suggestion
            urlh = self._request_webpage(
                HEADRequest(m3u8_url), video_id, 'Checking for HLS formats', 'No HLS formats found', fatal=False)
            ext = urlhandle_detect_ext(urlh) if urlh else None
```

Reply:
Still errors at that stage of the test... now with `ERROR: [ThePlatform] 0260338: [ThePlatform] No HLS formats found: HTTP Error 400: Bad Request`
I'm running out of steam for today. If you have any other ideas go ahead, otherwise I'll try to see if I can find a solution in the next few days.

Reply:
did you add `expected_warnings` to the test? https://github.com/yt-dlp/yt-dlp/pull/7484#discussion_r1278367646

Reply:
OK, I'm definitely running out of steam, the bit about changing the expected warnings only clicked now. I'll see if that does it.